src/StackDeploy/InstanceSpec.hs

```diff
@@ -80,5 +82,8 @@ mk name template = InstanceSpec
   , ..
   }
 
+mkName :: Text -> Name
+mkName = Provider.mkName
+
 templateProvider :: Provider -> Template.Provider
-templateProvider provider = template <$> provider
+templateProvider provider = fromList $ template <$> toList provider
```

Comment:
This could be written as:

```haskell
templateProvider = fromList . (template <$>) . toList
```

Reply:
I still do not "like" eta reduction that is not "primitive". But I really like the end result of your suggestion in terms of "fluency". 

I'll change to your suggestion on next touch.